test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1

```diff
@@ -0,0 +1,135 @@
+                    }
+                    elseif ($line.StartsWith("To: "))
+                    {
+                        if ($rv.To -eq $null)
+                        {
+                            $rv.To = @()
+                        }
+
+                        $rv.To += $line.Substring(4)
+                    }
+                    elseif ($line.StartsWith("Subject: "))
+                    {
+                        $rv.Subject = $line.Substring(9);
+                    }
+                    elseif ($line.Length -eq 0)
+                    {
+                        $state = "body"
+                    }
+                }
+                "body"
+                {
+                    if ($line.Length -eq 0)
+                    {
+                        $state = "done"
+                        continue
+                    }
+
+                    if ($rv.Body -eq $null)
+                    {
+                        $rv.Body = @()
+                    }
+
+                    $rv.Body += $line
+                }
+            }
+        }
+
+        return $rv
+    }
+
+    BeforeAll {
+        $PesterArgs = @{ Name = "Can send mail message from user to self"}
+        $alreadyHasMail = $true
+
+        if (-not $IsLinux)
+        {
+            $PesterArgs["Skip"] = $true
+            $PesterArgs["Name"] += " (skipped: not Linux)"
+            return
+        }
+
+        $user = "jeff"
+        $inPassword = Select-String "^${user}:" /etc/passwd -ErrorAction SilentlyContinue
+        if (-not $inPassword)
+        {
+            $PesterArgs["Pending"] = $true
+            $PesterArgs["Name"] += " (pending: user not in /etc/passwd)"
+            return
+        }
+
+        $domain = "Chinstrap"
+        if ($domain -ne [System.Environment]::MachineName)
+        {
+            $PesterArgs["Pending"] = $true
+            $PesterArgs["Name"] += " (pending: machine name not '$domain')"
+            return
+        }
+        $address = "$user@$domain"
+        $mailStore = "/var/mail"
+        $mailBox = Join-Path $mailStore $user
+        $mailBoxFile = Get-Item $mailBox -ErrorAction SilentlyContinue
+        if ($mailBoxFile -ne $null -and $mailBoxFile.Length -gt 2)
+        {
+            $PesterArgs["Pending"] = $true
+            $PesterArgs["Name"] += " (pending: mailbox not empty)"
+            return
+        }
+        $alreadyHasMail = $false
+        Set-Content -Value "" -Path $mailBox -Force -ErrorAction SilentlyContinue
+        $mailBoxFile = Get-Item $mailBox -ErrorAction SilentlyContinue
+        if ($mailBoxFile -eq $null -or $mailBoxFile.Length -gt 2)
+        {
+            $PesterArgs["Pending"] = $true
+            $PesterArgs["Name"] += " (pending: did not clear or create mailbox)"
+            return
+        }
+    }
+    AfterAll {
+       if (-not $alreadyHasMail)
+       {
+           Set-Content -Value "" -Path $mailBox -Force -ErrorAction SilentlyContinue
```

Comment:
Unless we're running as root or as a member of the `mail` group, we can write to the file but we cannot delete it.